core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -261,14 +261,48 @@ public static void createTable(
             int tableVersion,
             int tableId
     ) {
-        LOG.debug().$("create table [name=").$(tableDir).$(']').$();
-        path.of(root).concat(tableDir);
+        createTable(ff, root, mkDirMode, memory, path, false, tableDir, structure, tableVersion, tableId);
+    }
 
-        if (ff.mkdirs(path.slash$(), mkDirMode) != 0) {
-            throw CairoException.critical(ff.errno()).put("could not create [dir=").put(path).put(']');
+    public static void createTable(
+            FilesFacade ff,
+            CharSequence root,
+            int mkDirMode,
+            MemoryMARW memory,
+            Path path,
+            boolean pathIsLoadedWithVolume,
+            CharSequence tableDir,
+            TableStructure structure,
+            int tableVersion,
+            int tableId
+    ) {
+        Path normalPath = null;
+        if (!pathIsLoadedWithVolume) {
+            path.of(root).concat(tableDir).$();
+            LOG.info().$("create table [name=").utf8(tableDir).I$();
+        } else {
+            // path has been set by CREATE TABLE ... [IN VOLUME 'path'].
+            // it is a valid folder, or link to a folder, checked at bootstrap
+            normalPath = Path.getThreadLocal2(root).concat(tableDir).$();
```

Comment:
Don't we need to do `path.of(root).concat(tableDir).$()` in this branch to add table name to the volume path?

In general, I see no happy path tests for CREATE TABLE IN .. VOLUME DDL, only some direct calls to the underlying methods in `TableUtilsTest`. We need more tests for this feature.

Reply:
`ServerMainForeignTableTest` happy path.

The method ^ ^ has two operation modes, what we had, and the new mode. To differentiate between them we have a new boolean flag. The parameter path is passed to the method so that table utils does not need to have an instance, and in the normal mode we do not care what it has in it, we replace it with `root/table_name`. If however the boolean is true, then path is preloaded with the table's path. We will create the table there, however we will create a symlink in the normalPath.

Reply:
> `ServerMainForeignTableTest` happy path.

No, that's not the case. Try setting a break point on the `normalPath = Path.getThreadLocal2(root).concat(tableDir).$();` line and you'll see that it's never executed in `ServerMainForeignTableTest`. That's because the generated DDL contains an intermediate semicolon:
```
CREATE TABLE testServerMainCreateTableInVolume( investmentMill LONG, ticketThous INT, broker SYMBOL INDEX CAPACITY 32, ts TIMESTAMP) TIMESTAMP(ts) PARTITION BY DAY; IN VOLUME 'SECONDARY VOLUME'
```

The fact that the test doesn't do what it's supposed to be doing suggests that the assertions should be improved.

Reply:
checking...  OMG, you are right, improving this now

Reply:
this is next level awareness